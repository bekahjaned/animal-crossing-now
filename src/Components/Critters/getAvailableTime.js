const getAvailableTime = (hours) => {
    let availableTime = "";

    hours.forEach((hour) => {
      if (hour.length === 24) {
        availableTime = "all day";
      } else {
        const AmOrPm1 = hour[0] >= 12 ? "pm" : "am";
        hour[0] = hour[0] % 12 || 12;

        const AmOrPm2 = hour[hour.length - 1] >= 12 ? "pm" : "am";
        hour[hour.length - 1] = hour[hour.length - 1] % 12 || 12;

        if (availableTime !== "") {
          availableTime += ", ";
        }
        availableTime += `${hour[0] + AmOrPm1} - ${
          hour[hour.length - 1] + AmOrPm2
        }`;
      }
    });

    return availableTime;
}

module.exports = getAvailableTime;