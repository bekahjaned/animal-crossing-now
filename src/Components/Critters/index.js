import React from "react";

import BugCard from "../Cards/BugCard/index";
import FishCard from "../Cards/FishCard/index";

import { CritterDisplay } from "../../Elements/CritterDisplay/";
import { CritterGrid } from "../../Elements/CritterGrid/";

class Critters extends React.Component {
  render() {
    const availableFish = this.props.allAvailableFish.map((fish) => {
      return (
        <FishCard
          image={fish.image}
          name={fish.name}
          location={fish.location}
          key={fish.name}
        />
      );
    });

    const availableBugs = this.props.allAvailableBugs.map((bug) => {
      return (
        <BugCard
          image={bug.image}
          name={bug.name}
          location={bug.location}
          key={bug.name}
        />
      );
    });

    return (
      <div>
        <CritterDisplay>Fish available now</CritterDisplay>
        <CritterGrid>{availableFish}</CritterGrid>
        <CritterDisplay>Bugs available now</CritterDisplay>
        <CritterGrid>{availableBugs}</CritterGrid>
      </div>
    );
  }
}

export default Critters;
