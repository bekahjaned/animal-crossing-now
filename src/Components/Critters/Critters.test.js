import React from "react";
import renderer from "react-test-renderer";

import Critters from "./index";

it("renders multiple FishCards and BugCards", () => {
  // given
  const allAvailableFish = [
    {
      name: "Moray Eel",
      location: "Sea",
      image: "Moray Eel",
    },
    {
      name: "Giant Trevally",
      location: "Pier",
      image: "Giant Trevally",
    },
  ];

  const allAvailableBugs = [
    {
      name: "Ant",
      location: "On rotten food",
      image: "Ant",
    },
    {
      name: "Wharf Roach",
      location: "On beach rocks",
      image: "Wharf Roach",
    },
  ];

  //   when
  const tree = renderer
    .create(
      <Critters
        allAvailableFish={allAvailableFish}
        allAvailableBugs={allAvailableBugs}
      />
    )
    .toJSON();

  // then
  expect(tree).toMatchSnapshot();
});
