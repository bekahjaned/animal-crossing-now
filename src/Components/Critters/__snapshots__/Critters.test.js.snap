// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders multiple FishCards and BugCards 1`] = `
<div>
  <h3
    className="sc-AxjAm kbdyhu"
  >
    Fish available now
  </h3>
  <div
    className="sc-AxirZ bHoUSP"
  >
    <div
      className="critter"
    >
      <img
        alt="fish"
        src="Moray Eel.png"
      />
      <p
        className="critter-name"
      >
        Moray Eel
      </p>
      <p>
        Sea
      </p>
    </div>
    <div
      className="critter"
    >
      <img
        alt="fish"
        src="Giant Trevally.png"
      />
      <p
        className="critter-name"
      >
        Giant Trevally
      </p>
      <p>
        Pier
      </p>
    </div>
  </div>
  <h3
    className="sc-AxjAm kbdyhu"
  >
    Bugs available now
  </h3>
  <div
    className="sc-AxirZ bHoUSP"
  >
    <div
      className="critter"
    >
      <img
        alt="bug"
        src="Ant.png"
      />
      <p
        className="critter-name"
      >
        Ant
      </p>
      <p>
        On rotten food
      </p>
    </div>
    <div
      className="critter"
    >
      <img
        alt="bug"
        src="Wharf Roach.png"
      />
      <p
        className="critter-name"
      >
        Wharf Roach
      </p>
      <p>
        On beach rocks
      </p>
    </div>
  </div>
</div>
`;
